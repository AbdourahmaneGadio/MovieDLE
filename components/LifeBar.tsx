import { StyleSheet, View } from 'react-native';

type LifeBarProps = {
  lifePointsLost: number;
};

export default function LifeBar({
  lifePointsLost,
}: LifeBarProps) {
  const lifePoints = 100;
  const lifePointsRemaning =
    lifePoints - lifePointsLost;

  return (
    <View style={stylesLifeBar.container}>
      <View
        style={{
          backgroundColor: 'green',
          borderTopLeftRadius: 5,
          borderBottomLeftRadius: 5,
          borderTopRightRadius:
            lifePointsRemaning === 100 ? 5 : 0,
          borderBottomRightRadius:
            lifePointsRemaning === 100 ? 5 : 0,
          width: `${lifePointsRemaning}%`,
          visibility:
            lifePointsRemaning > 0
              ? ''
              : 'hidden',
        }}
        testID="remainingLife"></View>
      <View
        style={{
          backgroundColor: 'red',
          borderTopLeftRadius:
            lifePointsRemaning === 0 ? 5 : 0,
          borderBottomLeftRadius:
            lifePointsRemaning === 0 ? 5 : 0,
          borderTopRightRadius: 5,
          borderBottomRightRadius: 5,
          width: `${lifePointsLost}%`,
          visibility:
            lifePointsLost > 0 ? '' : 'hidden',
        }}
        testID="lifeLost"></View>
    </View>
  );
}

const stylesLifeBar = StyleSheet.create({
  container: {
    borderColor: 'gold',
    borderWidth: 5,
    marginVertical: 10,
    flexDirection: 'row',
    width: `90%`,
    borderRadius: 10,
    height: 40,
  },
});
