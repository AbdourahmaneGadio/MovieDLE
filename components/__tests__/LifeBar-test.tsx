import {
  cleanup,
  render,
} from '@testing-library/react-native';

import LifeBar from '@/components/LifeBar';

afterEach(cleanup);

describe('<LifeBar />', () => {
  test('LifeBar renders correctly', () => {
    const tree = render(
      <LifeBar lifePointsLost={0} />
    ).toJSON;
    expect(tree).toMatchSnapshot();
  });

  test('Remaining life should be full', () => {
    const { getByTestId } = render(
      <LifeBar lifePointsLost={0} />
    );

    const remainingLife = getByTestId(
      'remainingLife'
    );
    const lifeLost = getByTestId('lifeLost');

    expect(remainingLife).toBeOnTheScreen();
    expect(remainingLife).toHaveStyle({
      width: '100%',
    });
    expect(lifeLost).toHaveStyle({
      width: '0%',
      visibility: 'hidden',
    });
  });

  test('Life bar should be half full', () => {
    const { getByTestId } = render(
      <LifeBar lifePointsLost={50} />
    );

    const remainingLife = getByTestId(
      'remainingLife'
    );
    const lifeLost = getByTestId('lifeLost');

    expect(remainingLife).toBeOnTheScreen();
    expect(remainingLife).toHaveStyle({
      width: '50%',
    });
    expect(lifeLost).toHaveStyle({
      width: '50%',
      visibility: '',
    });
  });

  test('Life lost should be full', () => {
    const { getByTestId } = render(
      <LifeBar lifePointsLost={100} />
    );

    const remainingLife = getByTestId(
      'remainingLife'
    );
    const lifeLost = getByTestId('lifeLost');

    expect(remainingLife).toBeOnTheScreen();
    expect(remainingLife).toHaveStyle({
      width: '0%',
      visibility: 'hidden',
    });
    expect(lifeLost).toHaveStyle({
      width: '100%',
      visibility: '',
    });
  });
});
