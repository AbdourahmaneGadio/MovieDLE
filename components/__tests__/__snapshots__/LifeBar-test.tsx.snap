// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LifeBar /> LifeBar renders correctly 1`] = `[Function]`;
